### ISSUE TITLE
Docs: Clean up templates chapter

### ISSUE THREAD

**Issue by caponica** [sonata-project/SonataAdminBundle#1531]
Improve [this chapter](https://github.com/sonata-project/SonataAdminBundle/blob/master/Resources/doc/reference/templates.rst) and make it more accessible to newcomers by making the following changes:
- 12 (first section) "The default values will be set only if..." reword this to "The default values can be over-ridden by..." and then showing an example.
- can the default templates be set (in YML) for a specific XyzAdmin class or would that class need to define them using setTemplates (or is it getTemplates like the example shown? Or both?)
- 12.1 (Row template) edit for readability
- 12.1.1 (How to use it) does the example code render the screenshot? If so, how is the valid/invalid graphic rendered?
- The last paragraph could be improved with notes about how to avoid these pitfalls
- ensure that this page has full documentation for the sonata_admin.templates related config.yml settings so these can be removed from the "configuration" chapter - see #1522 for details

Note the section numbers mentioned are based on the [rendered docs for this chapter](http://sonata-project.org/bundles/admin/master/doc/reference/templates.html) on the sonata website.

**Comment by jordisala1991** [comment]
Template docs were improved on #1638. If you (or @sonata-project/contributors ) think that os not enough and the issue should remain opened, please ping me or reopen.

Closing
